README.md

```diff
@@ -1,31 +1,348 @@
 # Hyperion
 
-TODO: Write a gem description
+Hyperion is a Ruby REST client that follows certain conventions
+layered on top of HTTP. The conventions implement best practices
+surrounding versioning, error handling, and crafting an API for
+consumption by third parties. Hyperion provides abstractions that
+make it easy to follow these conventions consistently across
+projects.
 
-## Installation
+This document describes the conventions, then demonstrates how to
+use the API, and finally shows how to test your client-side code.
 
-Add this line to your application's Gemfile:
+## Conventions
+
+### Versioning
+
+_TODO_
+
+### Errors
+
+400-level errors are always returned as exactly 400; no distinction is
+made in the error code. Instead, a well-defined structure is returned
+that contains
+
+- a human-oriented error message, and
+- a machine-oriented list of `ErrorInfo`s.
+
+The point of the message is to describe the problem. The point of the
+error infos is to provide enough information to begin resolving the
+problem.
+
+An `ErrorInfo` consists of:
+
+- code
+- resource
+- field
+- value
+- reason
+
+As of this writing, allowable codes are
+
+- 'missing'
+- 'missing_field'
+- 'invalid'
+- 'unsupported'
+- 'already_exists'
+
+Each field is a string. Depending on the code, some fields may not
+apply. Inapplicable fields are always present, with the empty string
+as their value. This simplifies the code that deals with them.
+
+
+## Using hyperion
+
+Hyperion's API revolves around the idea of a _route_, which is a
+combination of:
+
+- HTTP method
+- URI
+- parameters that influence header generation and de/serialization
+  - `ResponseDescriptor` (what kind of data do we want back)
+    - influences the `Accept` header
+  - `PayloadDescriptor` (what kind of data is being PUT or POSTed)
+    - influences the `Content-Type` header
+
+Hyperion automatically deserializes responses according to the
+`ResponseDescriptor` and serializes the request payload according to
+the `PayloadDescriptor`.
+
+Hyperion provides a basic interface for requesting routes.
+
+```ruby
+require 'hyperion'
+
+route = RestRoute.new(:get, 'http://somesite.org/users/0', ResponseDescriptor.new('user', 1, :json))
```

Comment:
Based on the documentation so far, when you create a ResponseDescriptor, the `1` looks like a magic number.